common/static/common/js/components/views/paging_header.js

```diff
@@ -1,113 +1,64 @@
+                this.collection.bind('add', _.bind(this.render, this));
+                this.collection.bind('remove', _.bind(this.render, this));
+                this.collection.bind('reset', _.bind(this.render, this));
             },
 
-            messageHtml: function() {
-                var message = '';
-                var asset_type = false;
-                if (this.view.collection.assetType) {
-                    if (this.view.collection.sortDirection === 'asc') {
-                        // Translators: sample result:
-                        // "Showing 0-9 out of 25 total, filtered by Images, sorted by Date Added ascending"
-                        message = gettext('Showing %(current_item_range)s out of %(total_items_count)s, filtered by %(asset_type)s, sorted by %(sort_name)s ascending');
-                    } else {
-                        // Translators: sample result:
-                        // "Showing 0-9 out of 25 total, filtered by Images, sorted by Date Added descending"
-                        message = gettext('Showing %(current_item_range)s out of %(total_items_count)s, filtered by %(asset_type)s, sorted by %(sort_name)s descending');
-                    }
-                    asset_type = this.filterNameLabel();
+            render: function () {
+                var message,
+                    start = this.collection.start,
+                    end = start + this.collection.length,
+                    num_items = this.collection.totalCount;
+                if (!this.collection.hasPreviousPage() && !this.collection.hasNextPage()) {
+                    // One page of results
+                    message = interpolate(
+                        ngettext(
+                            /*
+                             * Translators: 'num_items' is the number of items that the student sees
+                             * 'item_display_name_singular' is the translated singular form of the item name
+                             * 'item_display_name_plural' is the translated plural form of the item name
+                             */
+                            'Currently viewing %(num_items)s %(item_display_name_singular)s',
+                            'Currently viewing all %(num_items)s %(item_display_name_plural)s',
+                            num_items
+                        ), {
+                            num_items: num_items,
+                            item_display_name_singular: this.itemDisplayNameSingular,
+                            item_display_name_plural: this.itemDisplayNamePlural
+                        }, true
+                    );
+                } else {
+                    // Many pages of results
+                    message = interpolate(
+                        ngettext(
+                            'Currently viewing %(first_index)s through %(last_index)s of %(num_items)s %(item_display_name_singular)s',
```

Comment:
I looked at our wording in Studio > Content > Files & Uploads. There might be some advantage to using the same wording as is used there. 
"Currently viewing" -> "Showing"
Instead of "through" just a dash for a range, e.g. "7-12"
They don't include a noun for the objects shown.
For example: "Showing 1-50 out of 120 total." 
If there is only one item, "Showing 1 out of 1 total."
If it's sorted, "Showing 1-50 out of 120 total, sorted by *Date Added* descending."
Keeps things simple and gets rid of the singular/plural issue too.
What do people think? @explorerleslie @dan-f @andy-armstrong